src/core/qgsgpkgflusher.cpp

```diff
@@ -90,8 +96,26 @@ void QgsGpkgFlusher::onLayersAdded( const QList<QgsMapLayer *> &layers )
   }
 }
 
+void QgsGpkgFlusher::stop()
+{
+  mFlusher->stop();
```

Comment:
This needs to be protected with a mutex and the same mutex needs to be used in flush().
Since `flush()` is run in a thread, stop can be called when a flush is already ongoing. But we want to be sure there is no flushing going on when `stop()` returns.

Reply:
you are very right, will fix this